tests/migrations/test_commands.py

```diff
@@ -1303,6 +1303,27 @@ class Meta:
             call_command("migrate", "migrated_app", "zero", verbosity=0)
             call_command("migrate", "migrated_unapplied_app", "zero", verbosity=0)
 
+    @override_settings(
+        INSTALLED_APPS=[
+            "migrations.migrations_test_apps.migrated_unapplied_app",
+            "migrations.migrations_test_apps.migrated_app",
+        ]
+    )
+    def test_migrate_not_reflected_changes_with_check(self):
+        class MigratedNewModelMigrated(models.Model):
+            class Meta:
+                app_label = "migrated_app"
+
+        class UnappliedNewModel(models.Model):
+            class Meta:
+                app_label = "migrated_unapplied_app"
+
+        out = io.StringIO()
+        with self.assertRaises(SystemExit):
+            call_command("migrate", stdout=out, no_color=True, check_unapplied=True)
+
+        self.assertEqual(out.getvalue(), "")
```

Comment:
This is not a regression test as it works without the patch. You need to first have a database that is up to date, e.g.
```python
    @override_settings(
        INSTALLED_APPS=[
            "migrations.migrations_test_apps.migrated_app",
        ],  
    )   
    def test_migrate_not_reflected_changes_with_check(self):
        out = io.StringIO()
        try:
            call_command("migrate", "migrated_app", verbosity=0)
            call_command("migrate", "migrated_app", stdout=out, no_color=True, check_unapplied=True)
            self.assertEqual(out.getvalue(), "")
        finally:                    
            call_command("migrate", "migrated_app", "zero", verbosity=0)

```

Reply:
I changed to regression test. and renamed.